pkg/reconciler/common/deployment.go

```diff
@@ -21,11 +21,26 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// IsDeploymentAvailable returns the whether the deployment is ready for the revision to be scaled up.
 func IsDeploymentAvailable(d *appsv1.Deployment) bool {
+	if d.Spec.Replicas == nil || *d.Spec.Replicas == 0 || d.Status.AvailableReplicas != *d.Spec.Replicas {
+		// If the Spec.Replicas is nil, or set to 0 due to any reasons for the revision to be scaled, or
+		// Status.AvailableReplicas is not equal to Spec.Replicas, we consider it NOT Available.
+		return false
+	}
+
 	for _, c := range d.Status.Conditions {
 		if c.Type == appsv1.DeploymentAvailable && c.Status == corev1.ConditionTrue {
 			return true
 		}
 	}
 	return false
 }
+
+// IsDeploymentHavingPods returns the whether the deployment has pods running.
+func IsDeploymentHavingPods(d *appsv1.Deployment) bool {
```

Comment:
Do you mean by healthy pods or include pods that could be in CLBO?

Reply:
No, AvailableReplicas in status means `how many pods are ready for at least some time`. They must be healthily running, not the pods in CLBO.